src/pages/home/report/ReportActionItem.js

```diff
@@ -95,101 +96,84 @@ const propTypes = {
-    }
+    const toggleContextMenuFromActiveReportAction = useCallback(() => {
+        setIsContextMenuActive(ReportActionContextMenu.isActiveReportAction(props.action.reportActionID));
+    }, [props.action.reportActionID]);
 
     /**
      * Show the ReportActionContextMenu modal popover.
      *
      * @param {Object} [event] - A press event.
      */
-    showPopover(event) {
-        // Block menu on the message being Edited or if the report action item has errors
-        if (this.props.draftMessage || !_.isEmpty(this.props.action.errors)) {
-            return;
-        }
-
-        this.setState({isContextMenuActive: true});
-
-        const selection = SelectionScraper.getCurrentSelection();
-        ReportActionContextMenu.showContextMenu(
-            ContextMenuActions.CONTEXT_MENU_TYPES.REPORT_ACTION,
-            event,
-            selection,
-            this.popoverAnchor,
-            this.props.report.reportID,
-            this.props.action,
-            this.props.draftMessage,
-            undefined,
-            this.checkIfContextMenuActive,
-            ReportUtils.isArchivedRoom(this.props.report),
-            ReportUtils.chatIncludesChronos(this.props.report),
-            this.props.action.childReportID,
-        );
-    }
+    const showPopover = useCallback(
+        (event) => {
+            // Block menu on the message being Edited or if the report action item has errors
+            if (props.draftMessage || !_.isEmpty(props.action.errors)) {
+                return;
+            }
+
+            setIsContextMenuActive(true);
+
+            const selection = SelectionScraper.getCurrentSelection();
+            ReportActionContextMenu.showContextMenu(
+                ContextMenuActions.CONTEXT_MENU_TYPES.REPORT_ACTION,
+                event,
+                selection,
+                popoverAnchorRef,
+                props.report.reportID,
+                props.action,
+                props.draftMessage,
+                () => {},
+                toggleContextMenuFromActiveReportAction,
+                ReportUtils.isArchivedRoom(props.report),
+                ReportUtils.chatIncludesChronos(props.report),
+                props.action.childReportID,
+            );
+        },
+        [props.draftMessage, props.action, props.report, toggleContextMenuFromActiveReportAction],
+    );
 
-    toggleReaction(emoji) {
-        Report.toggleEmojiReaction(this.props.report.reportID, this.props.action, emoji);
-    }
+    const toggleReaction = useCallback(
+        (emoji) => {
+            Report.toggleEmojiReaction(props.report.reportID, props.action, emoji);
+        },
+        [props.report, props.action],
+    );
 
     /**
      * Get the content of ReportActionItem
      * @param {Boolean} hovered whether the ReportActionItem is hovered
      * @returns {Object} child component(s)
      */
-    renderItemContent(hovered = false) {
+    const renderItemContent = (hovered = false) => {
```

Comment:
NAB but if this is just a regular function we could standardize on `function renderItemContent()`